py/picca/delta_extraction/expected_fluxes/true_continuum.py

```diff
@@ -0,0 +1,392 @@
+                                            fill_value="extrapolate")
+            self.get_stack_delta_weights = interp1d(
+                stack_log_lambda[stack_weight > 0.],
+                stack_weight[stack_weight > 0.],
+                kind="nearest",
+                fill_value=0.0,
+                bounds_error=False)
+        elif Forest.wave_solution == "lin":
+            self.get_stack_delta = interp1d(stack_lambda[stack_weight > 0.],
+                                            stack_delta[stack_weight > 0.],
+                                            kind="nearest",
+                                            fill_value="extrapolate")
+            self.get_stack_delta_weights = interp1d(
+                stack_lambda[stack_weight > 0.],
+                stack_weight[stack_weight > 0.],
+                kind="nearest",
+                fill_value=0.0,
+                bounds_error=False)
+        else:
+            raise ExpectedFluxError("Forest.wave_solution must be either "
+                                    "'log' or 'linear'")
+
+
+    def compute_expected_flux(self, forests, out_dir):
+        """
+
+        Arguments
+        ---------
+        forests: List of Forest
+        A list of Forest from which to compute the deltas.
+
+        out_dir: str
+        Directory where iteration statistics will be saved
+
+        Raise
+        -----
+        ExpectedFluxError if Forest.wave_solution is not 'lin' or 'log'
+        """
+        context = multiprocessing.get_context('fork')
+        for iteration in range(1):
+            pool = context.Pool(processes=self.num_processors)
+            self.logger.progress(
+                f"Continuum fitting: starting iteration {iteration} of {self.num_iterations}"
+            )
+
+            forests = pool.map(self.read_true_continuum, forests)
+            pool.close()
+
+
+        # now loop over forests to populate los_ids
+        self.compute_delta_stack(forests)
+        # now loop over forests to populate los_ids
+        self.populate_los_ids(forests)
+
+
+    def populate_los_ids(self, forests):
+        """Populate the dictionary los_ids with the mean expected flux, weights,
+        and inverse variance arrays for each line-of-sight.
+
+        Arguments
+        ---------
+        forests: List of Forest
+        A list of Forest from which to compute the deltas.
+        """
+        for forest in forests:
+            if forest.bad_continuum_reason is not None:
+                self.logger.info(f"Rejected forest with los_id {forest.los_id} "
+                                 f"due to {forest.bad_continuum_reason}")
+                continue
+            # get the variance functions and statistics
+            log_lambda = forest.log_lambda
+            stack_delta = self.get_stack_delta(log_lambda)
+            var_lss = self.get_var_lss(log_lambda)
+
+
+            mean_expected_flux = forest.continuum #* stack_delta ##not sure of this stack_delta 
```

Comment:
I'm not sure we need the stack_delta here...

Reply:
You are multiplying the true continuum by the mean expected flux, so I'd say you don't